bundles/framework/publisher2/view/PublisherSideBarHandler.js

```diff
@@ -289,9 +289,40 @@ class PublisherSidebarUIHandler extends StateHandler {
         </div>;
     }
 
+    getStatsGridPanelItem () {
+        return {
+            key: PANEL_STATSGRID_ID,
+            label: Oskari.getMsg('Publisher2', 'BasicView.statsgrid.label'),
+            children: this.renderStatsGridPanel()
+        };
+    }
+
+    toggleStatsGridPanel (visible) {
+        const newCollapseItems = this.getState().collapseItems
+            .filter(item => item.key !== PANEL_STATSGRID_ID)
+            .map(item => item);
+
+        if (visible) {
+            // this is empty when thematic maps was off by default. We gotta gather these tools somehow
+            const hasTools = this.statsGridPanelHandler.initTools();
+            if (hasTools) {
+                const statsGridPanelItem = this.getStatsGridPanelItem();
+                newCollapseItems.push(statsGridPanelItem);
+            }
+        }
+
+        this.updateState({
+            collapseItems: newCollapseItems
+        });
+    }
+
     updateStatsgridPanel () {
         const newCollapseItems = this.getState().collapseItems.map(item => item);
```

Comment:
It's basically this?
```suggestion
        const newCollapseItems = [...this.getState().collapseItems];
```

Reply:
had been using the map() - syntax throughout the file to achieve the same effect. Changed all of those to use spread.